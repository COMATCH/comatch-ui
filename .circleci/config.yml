# Javascript Node CircleCI 2.0 configuration file
defaults: &defaults
  docker:
    - image: circleci/node:12.0
  environment:
    NODE_PATH: src/
  resource_class: small
  working_directory: ~/repo
version: 2
jobs:
  checkout_code:
    <<: *defaults
    steps:
      - restore_cache:
          keys:
            - source-{{ .Branch }}-{{ .Revision }}
            - source-{{ .Branch }}-
            - source-

      - checkout

      - save_cache:
          key: source-{{ .Branch }}-{{ .Revision }}
          paths:
            - "~/repo"

  install_dependencies:
    <<: *defaults
    steps:
      - restore_cache:
          key: source-{{ .Branch }}-{{ .Revision }}
      - restore_cache:
          keys:
            - depens-{{ checksum "package.json" }}
            - depens-
      - run: yarn install
      - save_cache:
          paths:
            - node_modules
          key: depens-{{ checksum "package.json" }}
  test:
    <<: *defaults
    steps:
      - restore_cache:
          key: source-{{ .Branch }}-{{ .Revision }}
      - restore_cache:
          key: depens-{{ checksum "package.json" }}
      - run: yarn test

  lint:
    <<: *defaults
    steps:
      - restore_cache:
          key: source-{{ .Branch }}-{{ .Revision }}
      - restore_cache:
          key: depens-{{ checksum "package.json" }}
      - run: yarn lint

  build:
    <<: *defaults
    steps:
      - restore_cache:
          key: source-{{ .Branch }}-{{ .Revision }}
      - restore_cache:
          key: depens-{{ checksum "package.json" }}

      - run: yarn build

      - persist_to_workspace:
          root: ~/repo
          paths:
            - dist
            - package.json
workflows:
  version: 2
  test_lint_and_build:
    jobs:
      - checkout_code
      - install_dependencies:
          requires:
            - checkout_code
      - lint:
          requires:
            - install_dependencies
      - test:
          requires:
            - install_dependencies
            - lint
      - build:
          requires:
            - install_dependencies
            - lint
            - test
