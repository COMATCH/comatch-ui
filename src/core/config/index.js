export { languageCodesArray, languageCodesMap } from './languageCodes';
