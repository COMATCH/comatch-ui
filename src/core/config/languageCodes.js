export const languageCodesArray = ['en', 'de', 'fr'];

export const languageCodesMap = {
    ENGLISH: 'en',
    GERMAN: 'de',
    FRENCH: 'fr',
};
