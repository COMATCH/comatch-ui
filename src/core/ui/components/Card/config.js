const BACKGROUND_COLOR = {
    BLACK: 'black',
    WHITE: 'white',
    LIGHT_BLUE: 'lightBlue',
    PRIMARY: 'primary',
    LIGHT_PRIMARY: 'lightPrimary',
    SECONDARY: 'secondary',
    LIGHT_GRAY: 'lightGray',
    MID_GRAY: 'midGray',
    GRAY: 'gray',
    DARK_GRAY: 'darkGray',
    GREEN: 'green',
    LIGHT_GREEN: 'lightGreen',
    DARK_GREEN: 'darkGreen',
    RED: 'red',
    LIGHT_RED: 'lightRed',
    DARK_RED: 'darkRed',
    LIGHT_ORANGE: 'lightOrange',
    ORANGE: 'orange',
    DARK_ORANGE: 'darkOrange',
    YELLOW: 'yellow',
    TRANSPARENT: 'transparent',
};

const FONT_COLOR = {
    BLACK: 'white',
    WHITE: 'black',
    LIGHT_BLUE: 'black',
    PRIMARY: 'white',
    LIGHT_PRIMARY: 'black',
    SECONDARY: 'white',
    LIGHT_GRAY: 'black',
    MID_GRAY: 'black',
    GRAY: 'white',
    DARK_GRAY: 'white',
    GREEN: 'white',
    LIGHT_GREEN: 'black',
    DARK_GREEN: 'white',
    RED: 'white',
    LIGHT_RED: 'black',
    DARK_RED: 'white',
    LIGHT_ORANGE: 'black',
    ORANGE: 'black',
    DARK_ORANGE: 'white',
    YELLOW: 'black',
    TRANSPARENT: 'black',
};

const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
};

export { BACKGROUND_COLOR, FONT_COLOR, ORIENTATION };
