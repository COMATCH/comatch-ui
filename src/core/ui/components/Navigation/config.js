export const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
    RESPONSIVE: 'responsive',
};
