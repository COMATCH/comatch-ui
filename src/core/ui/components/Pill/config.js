import { palette, typography } from '../../../styles/variables';

const SIZES = {
    XS: 'XS',
    SM: 'SM',
    MD: 'MD',
    LG: 'LG',
    XL: 'XL',
    XXL: 'XXL',
};

const SIZES_IN_PIXELS = {
    XS: {
        fontSize: typography.fontXs,
        lineHeight: typography.lineHeightXs,
    },
    SM: {
        fontSize: typography.fontSm,
        lineHeight: typography.lineHeightSm,
    },
    MD: {
        fontSize: typography.fontMd,
        lineHeight: typography.lineHeightMd,
    },
    LG: {
        fontSize: typography.fontLg,
        lineHeight: typography.lineHeightLg,
    },
    XL: {
        fontSize: typography.fontXl,
        lineHeight: typography.lineHeightXl,
    },
    XXL: {
        fontSize: typography.fontXxl,
        lineHeight: typography.lineHeightXxl,
    },
};

const TYPES = {
    DISABLED: 'DISABLED',
    ERROR: 'ERROR',
    INFO: 'INFO',
    PRIMARY: 'PRIMARY',
    SECONDARY: 'SECONDARY',
    SUCCESS: 'SUCCESS',
    WARN: 'WARN',
};

const TYPES_IN_COLOR = {
    DISABLED: {
        background: palette.lightGray,
        borderColor: palette.lightGray,
        color: palette.black,
    },
    ERROR: {
        background: palette.red,
        borderColor: palette.red,
        color: palette.white,
    },
    INFO: {
        background: palette.lightBlue,
        borderColor: palette.lightBlue,
        color: palette.black,
    },
    PRIMARY: {
        background: palette.primary,
        borderColor: palette.primary,
        color: palette.white,
    },
    SECONDARY: {
        background: palette.secondary,
        borderColor: palette.secondary,
        color: palette.white,
    },
    SUCCESS: {
        background: palette.green,
        borderColor: palette.green,
        color: palette.white,
    },
    WARN: {
        background: palette.orange,
        borderColor: palette.orange,
        color: palette.white,
    },
};

export { SIZES, SIZES_IN_PIXELS, TYPES, TYPES_IN_COLOR };
