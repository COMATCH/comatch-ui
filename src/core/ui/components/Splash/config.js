import { typography } from '../../../styles/variables';

const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
};

const SIZE = {
    XS: 'xs',
    SM: 'sm',
    MD: 'md',
    LG: 'lg',
    XL: 'xl',
    XXL: 'xxl',
};

const SIZE_TO_PIXELS = {
    xs: {
        title: `font-size: calc(${typography.fontXs} + 6px); line-height: calc(${typography.lineHeightXs} + 6px);`,
        subtitle: `font-size: ${typography.fontXs}; line-height: ${typography.fontXs};`,
    },
    sm: {
        title: `font-size: calc(${typography.fontSm} + 6px); line-height: calc(${typography.lineHeightSm} + 6px);`,
        subtitle: `font-size: ${typography.fontSm}; line-height: ${typography.fontSm};`,
    },
    md: {
        title: `font-size: calc(${typography.fontMd} + 6px); line-height: calc(${typography.lineHeightMd} + 6px);`,
        subtitle: `font-size: ${typography.fontMd}; line-height: ${typography.fontMd};`,
    },
    lg: {
        title: `font-size: calc(${typography.fontLg} + 6px); line-height: calc(${typography.lineHeightLg} + 6px);`,
        subtitle: `font-size: ${typography.fontLg}; line-height: ${typography.fontLg};`,
    },
    xl: {
        title: `font-size: calc(${typography.fontXl} + 6px); line-height: calc(${typography.lineHeightXl} + 6px);`,
        subtitle: `font-size: ${typography.fontXl}; line-height: ${typography.fontXl};`,
    },
    xxl: {
        title: `font-size: calc(${typography.fontXxl} + 6px); line-height: calc(${typography.lineHeightXxl} + 6px);`,
        subtitle: `font-size: ${typography.fontXxl}; line-height: ${typography.fontXxl};`,
    },
};

export { ORIENTATION, SIZE, SIZE_TO_PIXELS };
