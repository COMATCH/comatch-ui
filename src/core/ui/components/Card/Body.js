import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StyledBody } from './StyledWrapper';
import { BACKGROUND_COLOR, ORIENTATION } from './config';

const propTypes = {
    backgroundColor: PropTypes.oneOf(Object.keys(BACKGROUND_COLOR)),
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string,
    orientation: PropTypes.oneOf(Object.values(ORIENTATION)),
    wrap: PropTypes.bool,
};

const defaultProps = {
    backgroundColor: 'TRANSPARENT',
    children: null,
    className: null,
    id: null,
    orientation: ORIENTATION.VERTICAL,
    wrap: false,
};

const Body = ({ backgroundColor, children, className, id, orientation, wrap }) =>
    !!children && (
        <StyledBody
            backgroundColor={backgroundColor}
            className={classnames('Body', className)}
            {...(id && { id })}
            orientation={orientation}
            wrap={wrap}
        >
            {children}
        </StyledBody>
    );

Body.propTypes = propTypes;
Body.defaultProps = defaultProps;

export { Body };
