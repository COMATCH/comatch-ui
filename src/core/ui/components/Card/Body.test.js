import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Body } from './Body';
import { BACKGROUND_COLOR, ORIENTATION } from './config';
import '../../../config/tests/setup';

describe('Card.Body', () => {
    it('should render the default Body correctly', () => {
        const body = renderer.create(<Body>body</Body>).toJSON();
        expect(body).toMatchSnapshot();
    });

    it('should render a customized Body correctly', () => {
        const body = renderer
            .create(
                <Body
                    backgroundColor={BACKGROUND_COLOR.PRIMARY}
                    className="ABody"
                    id="a-body"
                    orientation={ORIENTATION.HORIZONTAL}
                    wrap
                >
                    <div>some content...</div>
                </Body>,
            )
            .toJSON();
        expect(body).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const body = mount(<Body />);
        expect(body.prop('backgroundColor')).toEqual('transparent');
        expect(body.prop('className')).toEqual(null);
        expect(body.prop('id')).toEqual(null);
        expect(body.prop('orientation')).toEqual('vertical');
        expect(body.prop('wrap')).toEqual(false);
    });
});
