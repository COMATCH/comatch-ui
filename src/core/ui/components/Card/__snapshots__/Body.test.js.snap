// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.Body should render a customized Body correctly 1`] = `
<div
  className="Body ABody sc-bdVaJa lcCQJK"
  id="a-body"
>
  <div>
    some content...
  </div>
</div>
`;

exports[`Card.Body should render the default Body correctly 1`] = `
<div
  className="Body sc-bdVaJa jJZBvK"
>
  body
</div>
`;
